velox/exec/tests/RowContainerTest.cpp

```diff
@@ -624,6 +624,92 @@ TEST_F(RowContainerTest, types) {
   EXPECT_LT(0, free.second);
 }
 
+TEST_F(RowContainerTest, extractNulls) {
+  constexpr int32_t kNumRows = 100;
+  auto batch = makeRowVector({
+      makeFlatVector<bool>(
+          kNumRows, [](auto row) { return row % 5 == 0; }, nullEvery(2)),
+      makeFlatVector<int8_t>(
+          kNumRows, [](auto row) { return row % 5; }, nullEvery(3)),
+      makeFlatVector<int16_t>(
+          kNumRows, [](auto row) { return row % 5; }, nullEvery(4)),
+      makeFlatVector<int32_t>(
+          kNumRows, [](auto row) { return row % 5; }, nullEvery(5)),
+      makeFlatVector<int64_t>(
+          kNumRows, [](auto row) { return row % 5; }, nullEvery(6)),
+      makeFlatVector<float>(
+          kNumRows, [](auto row) { return row % 5; }, nullEvery(7)),
+      makeFlatVector<double>(
+          kNumRows, [](auto row) { return row % 5; }, nullEvery(8)),
+      makeFlatVector<StringView>(
+          kNumRows, [](auto row) { return StringView("abcd"); }, nullEvery(9)),
+      makeArrayVector<int32_t>(
+          kNumRows,
+          [](auto i) { return i % 5; },
+          [](auto i) { return i % 10; },
+          nullEvery(10)),
+      makeMapVector<int32_t, int32_t>(
+          kNumRows,
+          [](auto i) { return i % 5; },
+          [](auto i) { return i % 10; },
+          [](auto i) { return i % 3; },
+          nullEvery(11)),
+      makeRowVector(
+          {"c0", "c1"},
+          {makeFlatVector<int32_t>(kNumRows, [](auto i) { return i % 5; }),
+           makeFlatVector<int32_t>(kNumRows, [](auto i) { return i % 7; })},
+          nullEvery(12)),
+  });
+
+  std::vector<TypePtr> rowType = {
+      BOOLEAN(),
+      TINYINT(),
+      SMALLINT(),
+      INTEGER(),
+      BIGINT(),
+      REAL(),
+      DOUBLE(),
+      VARCHAR(),
+      ARRAY(INTEGER()),
+      MAP(INTEGER(), INTEGER()),
+      ROW({INTEGER(), INTEGER()})};
+  auto data = makeRowContainer({}, rowType);
+  for (int i = 0; i < kNumRows; i++) {
+    data->newRow();
+  }
+
+  std::vector<char*> rows(kNumRows);
+  RowContainerIterator iter;
+  EXPECT_EQ(data->listRows(&iter, kNumRows, rows.data()), kNumRows);
+  SelectivityVector allRows(kNumRows);
+  for (int i = 0; i < batch->childrenSize(); i++) {
+    DecodedVector decoded(*batch->childAt(i), allRows);
+    for (auto j = 0; j < kNumRows; ++j) {
+      data->store(decoded, j, rows[j], i);
+    }
+  }
+
+  auto nulls = AlignedBuffer::allocate<bool>(kNumRows, pool());
+  auto expected = AlignedBuffer::allocate<bool>(kNumRows, pool());
+  auto rawExpected = expected->asMutable<uint64_t>();
+  for (int i = 0; i < 11; i++) {
+    data->extractNulls(rows.data(), kNumRows, i, nulls);
+
+    auto nullEvery = i + 2;
+    for (int j = 0; j < kNumRows; j += 1) {
```

Comment:
There is no need to build 'expected' buffer. Just loop over rows and assert each bit in 'nulls'.